nixos/modules/services/web-apps/mattermost.nix

```diff
@@ -44,7 +43,7 @@ let
     }) cfg.plugins;
 
   mattermostPlugins = with pkgs;
-    if mattermostPluginDerivations == null then null
+    if mattermostPluginDerivations == [] then null
```

Comment:
Didn't test it but could this condition not be removed? (At the cost of building a probably empty derivation). Or does this not allow manual plugin install or something else then?

Reply:
As far as I understand the code this is probably the reason.

Reply:
Manual plugin installs don't work for plugins that have binaries, since the dynamic linker may not be set properly in them, and we have to run autoPatchelf as part of the derivation build to make them work. That being said, I think there's a way to support both manual and Nix-installed plugins if we symlink in all the plugins when the server starts

Reply:
> That being said, I think there's a way to support both manual and Nix-installed plugins if we symlink in all the plugins when the server starts
I don't think we need to implement this for now. Could always be done later.